### PR TITLE
lint: convert spellchecking lint test to python

### PR DESCRIPTION
**Pull request by fjahr**
The new python version should produce the exact same output as the bash version but be easier to maintain.

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK - can you rebase now that #24762 has been merged.

File: test/lint/lint-spelling.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+"""
+
+from subprocess import check_output, STDOUT, CalledProcessError
+
+ignore_words_file = 'test/lint/spelling.ignore-words.txt'
+files_args = r'git ls-files -- ":(exclude)build-aux/m4/" ":(exclude)contrib/seeds/*.txt" ":(exclude)depends/" ":(exclude)doc/release-notes/" ":(exclude)src/leveldb/" ":(exclude)src/crc32c/" ":(exclude)src/qt/locale/" ":(exclude)src/qt/*.qrc" ":(exclude)src/secp256k1/" ":(exclude)src/minisketch/" ":(exclude)src/univalue/" ":(exclude)contrib/builder-keys/keys.txt" ":(exclude)contrib/guix/patches"'
+
+
+def main():
+    files = check_output(files_args, shell=True).decode("utf-8").splitlines()
```
Comment (laanwj) [inline test/lint/lint-spelling.py:19]: Please don't use `shell=true` if you can avoid it

**Comment by laanwj** [comment]
Concept ACK, thanks!

**Comment by theStack** [comment]
Concept ACK

**Review by brunoerg — COMMENTED** [review]
Concept ACK

File: test/lint/lint-spelling.py (hunk 1 above)
Comment (fjahr) [reply test/lint/lint-spelling.py]: was able to avoid it, done

**Comment by fjahr** [comment]
Rebased and added the error handling in case codespell is not installed which I had omitted in the last push and also avoided usage of `shell=True`.

**Comment by MarcoFalke** [comment]
cr ACK 4685463301a1c64c1be07725059cc94d69db104b

**Review by maflcko — APPROVED** [review]

File: test/lint/lint-spelling.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def main():
+    check_codespell_install()
+
+    files = check_output(FILES_ARGS).decode("utf-8").splitlines()
+    codespell_args = ['codespell', '--check-filenames', '--disable-colors', '--quiet-level=7', '--ignore-words={}'.format(IGNORE_WORDS_FILE)] + files
+
+    try:
+        check_output(codespell_args, stderr=STDOUT)
```
Comment (MarcoFalke) [inline test/lint/lint-spelling.py:33]: Will this print the spelling mistakes?

Comment (fjahr) [reply test/lint/lint-spelling.py:33]: For me it does but testing on more platforms might be a good idea :)

```
$ test/lint/lint-spelling.py
configure.ac:364: overriden ==> overridden
configure.ac:860: overriden ==> overridden
src/wallet/coinselection.h:228: Chooose ==> Choose
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/spelling.ignore-words.txt
```

Comment (MarcoFalke) [reply test/lint/lint-spelling.py:33]: Yeah, it should. I just missed the `print(e.output.decode` line below. :man_facepalming:

File: test/lint/lint-spelling.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+Warn in case of spelling errors.
+Note: Will exit successfully regardless of spelling errors.
+"""
+
+from subprocess import check_output, STDOUT, CalledProcessError
```
Comment (MarcoFalke) [inline test/lint/lint-spelling.py:12]: nit for future rewrites. I slightly prefer `import subprocess` and then `subprocess.symbol` in the code below. It is more verbose but clearer.

File: test/lint/lint-spelling.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+IGNORE_WORDS_FILE = 'test/lint/spelling.ignore-words.txt'
+FILES_ARGS = ['git', 'ls-files', '--', ":(exclude)build-aux/m4/", ":(exclude)contrib/seeds/*.txt", ":(exclude)depends/", ":(exclude)doc/release-notes/", ":(exclude)src/leveldb/", ":(exclude)src/crc32c/", ":(exclude)src/qt/locale/", ":(exclude)src/qt/*.qrc", ":(exclude)src/secp256k1/", ":(exclude)src/minisketch/", ":(exclude)src/univalue/", ":(exclude)contrib/builder-keys/keys.txt", ":(exclude)contrib/guix/patches"]
```
Comment (MarcoFalke) [inline test/lint/lint-spelling.py:15]: nit: Add trailing `,` and put each token on a newline to make editing and reading easier?

File: test/lint/lint-spelling.py (hunk 2 above)
Comment (MarcoFalke) [reply test/lint/lint-spelling.py:33]: nit: I guess an alternative would be to use `subprocess.check_call` and not redirect the output at all or `print()` it?

File: test/lint/lint-spelling.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    check_codespell_install()
+
+    files = check_output(FILES_ARGS).decode("utf-8").splitlines()
+    codespell_args = ['codespell', '--check-filenames', '--disable-colors', '--quiet-level=7', '--ignore-words={}'.format(IGNORE_WORDS_FILE)] + files
+
+    try:
+        check_output(codespell_args, stderr=STDOUT)
+    except CalledProcessError as e:
+        print(e.output.decode("utf-8"), end="")
+        print('^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in {}'.format(IGNORE_WORDS_FILE))
```
Comment (MarcoFalke) [inline test/lint/lint-spelling.py:36]: nit: Can use f-strings in new code

**Review by maflcko — COMMENTED** [review]
Left some nits for your future transpiles, feel free to ignore.

**Review by jonatack — COMMENTED** [review]
ACK, working for me on debian.

**Pull request merged**